packages/liquidator/index.js

```diff
@@ -53,24 +61,52 @@ async function run({
     const getTime = () => Math.round(new Date().getTime() / 1000);
 
     // Load unlocked web3 accounts and get the networkId.
-    const [accounts, networkId] = await Promise.all([web3.eth.getAccounts(), web3.eth.net.getId()]);
+    const [detectedContract, accounts, networkId] = await Promise.all([
+      findContractVersion(empAddress, web3),
+      web3.eth.getAccounts(),
+      web3.eth.net.getId()
+    ]);
+
+    // Append the contract version and type to the liquidatorConfig, if the liquidatorConfig does not already contain one.
+    if (!liquidatorConfig) liquidatorConfig = {};
+    if (!liquidatorConfig.contractVersion) liquidatorConfig.contractVersion = detectedContract.contractVersion;
+    if (!liquidatorConfig.contractType) liquidatorConfig.contractType = detectedContract.contractType;
+
+    // Check that the version and type is supported. Note if either is null this check will also catch it.
+    if (
+      SUPPORTED_CONTRACT_VERSIONS.filter(
+        vo => vo.contractType == liquidatorConfig.contractType && vo.contractVersion == liquidatorConfig.contractVersion
+      ).length == 0
+    )
+      throw new Error(
+        `Contract version specified or inferred is not supported by this bot. Loaded/inferred contractVersion:${
+          liquidatorConfig.contractVersion
+        } & contractType:${liquidatorConfig.contractType} is not part of ${JSON.stringify(SUPPORTED_CONTRACT_VERSIONS)}`
+      );
 
-    // Setup contract instances.
-    const voting = new web3.eth.Contract(getAbi("Voting"), getAddress("Voting", networkId));
-    const emp = new web3.eth.Contract(getAbi("ExpiringMultiParty"), empAddress);
+    // Setup contract instances. This uses the contract version pulled in from previous step. Voting is hardcoded to latest main net version.
+    const voting = new web3.eth.Contract(getAbi("Voting", "1.2.2"), getAddress("Voting", networkId));
+    const emp = new web3.eth.Contract(
```

Comment:
here we pull in the right contract type and version

Reply:
Do you forsee any issues where voting version and EMP version are different? I think that's not a problem but its  just weird to see the version hardcoded here for `voting` but not for `EMP`

Reply:
Well, the whole point of this change is that we _know_ the canonical deployed version of the voting contract (1.2.2.) and we can update this hardcoded version when we roll out a new release for voting. On the other hand, the EMP could be any version.

Perhaps we should have a common file that indicates the current production versioning of `Voting` that can then be pulled in but for now, I think this is easy enough.

Reply:
>Perhaps we should have a common file that indicates the current production versioning of Voting that can then be pulled in but for now, I think this is easy enough.

I think having some common dictionary listing all contract versions (or in the Voting's case, the one canonical version) would be useful. I can imagine that the voting dApp and other bots would read from this.